robot/EDA/resources/CourseConnectionsSettingsPageObject.py

```diff
@@ -20,3 +22,62 @@ def _is_current_page(self):
             locator_tab,
             error=f"Course Connections tab with locator '{locator_tab}' is not available on the page"
         )
+
+    def set_enable_course_connections(self):
+        """ Set the checkbox for 'Enable Course Connections' field """
+        locator = eda_lex_locators["eda_settings_cc"]["enable_cc_checkbox"]
+        self.selenium.wait_until_page_contains_element(locator)
+        self.selenium.click_element(locator)
+        time.sleep(0.5)
```

Comment:
Did you try waiting for a specific action, element, or class instead of time.sleep?